plugins/support-helphub/inc/helphub-customroles/includes/class-helphub-custom-roles.php

```diff
@@ -0,0 +1,246 @@
+<?php
+/**
+ * Class file of HelpHub custom roles.
+ *
+ * @package HelpHub
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class of HelpHub custom roles.
+ */
+class HelpHub_Custom_Roles {
+
+
+	/**
+	 * The single instance of HelpHub_Custom_Roles.
+	 *
+	 * @var     object
+	 * @access  private
+	 * @since   1.0.0
+	 */
+	private static $_instance = null;
+
+	/**
+	 * Settings class object
+	 *
+	 * @var     object
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $settings = null;
+
+	/**
+	 * The version number.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $_version;
+
+	/**
+	 * The token.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $_token;
+
+	/**
+	 * The main plugin file.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $file;
+
+	/**
+	 * The main plugin directory.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $dir;
+
+	/**
+	 * The plugin assets directory.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $assets_dir;
+
+	/**
+	 * The plugin assets URL.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $assets_url;
+
+	/**
+	 * Suffix for Javascripts.
+	 *
+	 * @var     string
+	 * @access  public
+	 * @since   1.0.0
+	 */
+	public $script_suffix;
+
+	/**
+	 * Custom roles Constructor.
+	 *
+	 * @param string $file    filename.
+	 * @param string $version version.
+	 */
+	public function __construct( $file = '', $version = '1.0.0' ) {
+		$this->_version = $version;
+		$this->_token   = 'helphub_custom_roles';
+
+		$this->file = $file;
+		$this->dir  = dirname( $this->file );
+
+		register_activation_hook( $this->file, array( $this, 'install' ) );
+
+		$this->add_helphub_customrole();
+
+		$this->load_plugin_textdomain();
+
+		add_action( 'admin_init', array( $this, 'hh_restrict_admin_pages' ), 0 );
+
+		add_action( 'init', array( $this, 'load_localisation' ), 0 );
+	} // End __construct ()
+
+	/**
+	 * This will add restriction to the custom rule.
+	 */
+	public function hh_restrict_admin_pages() {
+		if ( current_user_can( 'helphub_editor' ) ) {
```

Comment:
Might make more sense to check the users role, `in_array( 'helphub_editor', $user->roles )`, since it's a singular role (at this time at least) that shouldn't be touching the themes.